## Prep Dataset

In [3]:
# Install packages, load libraries.
library(tidyverse)
library(stringr)

── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 3.1.0     ✔ purrr   0.2.5
✔ tibble  2.0.1     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


Tidyverse is used in this notebook mostly for data manipulation and visualization. Relevant documentation is at https://dplyr.tidyverse.org and https://ggplot2.tidyverse.org.

Stringr used to recode values for some variables (e.g., North American Occupation).

In [19]:
# Converts all factors to character class
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) df[[i]] = as.character(df[[i]])
  return(df)
}
# Code from user "By0" at https://stackoverflow.com/questions/2851015/convert-data-frame-columns-from-factors-to-characters (line 14)

In [20]:
# Import, unfactorize and view data.
authorData <- unfactorize(read.csv("IMLD_AUTHORS_QA completed.csv", na.strings=c("", "Not indicated")))
docData <- unfactorize(read.csv("IMLD_DOCS_QA completed.csv", na.strings=c("", "Not indicated")))
glimpse(authorData)
glimpse(docData)

Observations: 2,162
Variables: 30
$ sourceids                 <chr> NA, "S10000; S9527", "S10001", "S10002", "S…
$ numdocs                   <int> 502, 23, 19, 23, 172, 2, 1, 3, 1, 1, 24, 1,…
$ docauthorid               <chr> "per0002637", "per0021589", "per0022935", "…
$ docauthorname             <chr> "Editor", "Pilibosian, Khachadoor, 1904-198…
$ alternatenames            <chr> NA, NA, NA, NA, "Giesberg, Henriette Ann El…
$ briefname                 <chr> "Editor", "Khachadoor Pilibosian", "Evelio …
$ authrace                  <chr> "Not applicable", "White", "Black", "White"…
$ nationality               <chr> "Not applicable", NA, "United States", "Uni…
$ religion                  <chr> "Not applicable", "Catholic; Christian", "C…
$ birthyear                 <int> NA, 1904, 1919, 1882, 1813, 1774, 1794, 181…
$ birthmonth                <int> NA, NA, NA, 4, NA, 8, NA, 9, 5, 4, NA, NA, …
$ birthday                  <int> NA, NA, NA, 16, NA, 11, NA, 17, 3, 23, NA, …
$ deathyear       

In [21]:
#Merge datasets
df <- right_join(authorData, docData, by = 'docauthorid')
glimpse(df)

Observations: 8,749
Variables: 100
$ sourceids                   <chr> NA, "S10000; S9527", "S10000; S9527", "S1…
$ numdocs                     <int> 502, 23, 23, 23, 23, 23, 23, 23, 23, 23, …
$ docauthorid                 <chr> "per0002637", "per0021589", "per0021589",…
$ docauthorname.x             <chr> "Editor", "Pilibosian, Khachadoor, 1904-1…
$ alternatenames              <chr> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ briefname.x                 <chr> "Editor", "Khachadoor Pilibosian", "Khach…
$ authrace.x                  <chr> "Not applicable", "White", "White", "Whit…
$ nationality.x               <chr> "Not applicable", NA, NA, NA, NA, NA, NA,…
$ religion.x                  <chr> "Not applicable", "Catholic; Christian", …
$ birthyear.x                 <int> NA, 1904, 1904, 1904, 1904, 1904, 1904, 1…
$ birthmonth.x                <int> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ birthday.x                  <int> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ deathyear.x    

In [22]:
# Converts character to factor class
factorize <- function(df){
  for(i in which(sapply(df, class) == "character")) df[[i]] = as.factor(df[[i]])
  return(df)
}

In [23]:
df <- factorize(df)
summary(df)

               sourceids       numdocs           docauthorid  
 S6210              : 232   Min.   :  1.00   per0002637: 460  
 S11073             : 189   1st Qu.: 10.00   per0038009: 190  
 S10003             : 180   Median : 28.00   per0022938: 172  
 S9140; S9435       : 171   Mean   : 69.89   per0018089: 171  
 S2344; S4963; S9924: 150   3rd Qu.: 64.00   per0004772: 150  
 (Other)            :7174   Max.   :502.00   per0007475: 149  
 NA's               : 653   NA's   :193      (Other)   :7457  
                               docauthorname.x
 Editor                                : 460  
 Harris, Sarah Stretch, 1818-1897      : 190  
 Bruns, Jette, 1813-1899               : 172  
 Berenson, Bernard, 1865-1959          : 171  
 Moodie, Susannah Strickland, 1803-1885: 150  
 (Other)                               :7413  
 NA's                                  : 193  
                                                       alternatenames
 Giesberg, Henriette Ann Elisabeth; Bruns, Mrs. Jo

## Select Columns

In [24]:
df  <- select(df,
       docsequence, 
       docid, 
       docyear, 
       doctype, 
       allsubject, 
       broadsubj, 
       personalevent,
       wwritten,
       docauthorid,
       docauthorname.x,
       docauthorname.y,
       language, 
       editor, 
       briefname.x,
       briefname.y, 
       authrace.x,
       authrace.y,
       nationality.x,
       nationality.y,
       religion.x,
       religion.y,
       birthyear.x,
       birthyear.y,
       deathyear.x,
       deathyear.y,
       birthplace.x,
       birthplace.y,
       deathplace.x,
       deathplace.y,
       ageatdeath, 
       authorgender.x,
       authorgender.y,
       native_occupation.x,
       native_occupation.y,
       north_american_occupation.x,
       north_american_occupation.y,
       year_immigration.x,
       year_immigration.y,
       cultural_heritage.x,
       cultural_heritage.y,
       stayed_north_america.x,
       stayed_north_america.y,
       author_generation.x, 
       author_generation.y, 
       agewriting,
       marriagestatus, 
       maternalstatus)
summary(df)

  docsequence             docid         docyear                 doctype    
 Min.   :  0.00   S10000-D001:   1   Min.   :1784   Chapter         :3367  
 1st Qu.:  4.00   S10000-D002:   1   1st Qu.:1880   Letter          :1866  
 Median : 14.00   S10000-D003:   1   Median :1925   Oral history    :1028  
 Mean   : 28.87   S10000-D004:   1   Mean   :1922   Diary           : 750  
 3rd Qu.: 35.00   S10000-D005:   1   3rd Qu.:1961   Emigration guide: 705  
 Max.   :241.00   S10000-D006:   1   Max.   :2004   Editorial       : 499  
                  (Other)    :8743   NA's   :168    (Other)         : 534  
                                                                allsubject  
 Correspondence; Intellectual life                                   :  13  
 Internment camps; Politics                                          :   9  
 Angel Island Immigration Center, San Francisco, CA; Meals; Health   :   7  
 Immigrants and emigrants; Irish people; Domestic life; Ethnic groups:   7  
 Immigr

In [25]:
# Create a new column for resolving author_generation discrepancy
df$author_generation <- df$author_generation.x
summary(df$author_generation)
summary(df$author_generation.x)
summary(df$author_generation.y)

First Not applicable          Other         Second           NA's 
          6025            735             35            625           1329

First Not applicable          Other         Second           NA's 
          6025            735             35            625           1329

First Not applicable          Other         Second           NA's 
          6054            735             35            625           1300

## Compare Subsets

In [26]:
#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, 
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation.x == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, 
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation.y == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 1225

[1] 1272

In [27]:
# What's the difference?
1272-1225

[1] 47

## Diagnostics

In [28]:
# Went back to check on 19 errors between the two datasets. 
# Ids selected from "Author Generation" output in CrosscheckData notebook.
authorData[authorData$docauthorid == "per0022893",]
unique(docData[docData$docauthorid == "per0022893", "doctype"])

authorData[authorData$docauthorid == "per0003974",]
unique(docData[docData$docauthorid == "per0003974", "doctype"])

sourceids numdocs docauthorid docauthorname               alternatenames
1989 S9955     18      per0022893  Clayton, William, 1814-1879 NA            
     briefname       authrace nationality   religion          birthyear ⋯
1989 William Clayton White    United States Mormon; Christian 1814      ⋯
     authorgender native_occupation north_american_occupation
1989 M            NA                NA                       
     organization_affiliations year_immigration cultural_heritage
1989 NA                        NA               English; European
     point_of_entry point_of_emigration stayed_north_america author_generation
1989 NA             NA                  Stayed               First

[1] "Diary"

sourceids numdocs docauthorid docauthorname               alternatenames
1999 S9974     1       per0003974  Carnegie, Andrew, 1835-1919 NA            
     briefname       authrace nationality   religion birthyear ⋯ authorgender
1999 Andrew Carnegie White    United States NA       1835      ⋯ M           
     native_occupation north_american_occupation organization_affiliations
1999 NA                NA                        NA                       
     year_immigration cultural_heritage  point_of_entry point_of_emigration
1999 NA               Scottish; European NA             NA                 
     stayed_north_america author_generation
1999 Stayed               First

[1] "Chapter"

These were not letters, so 19 discrepancies caught by arsenal do not account for the difference found here. 

According to right_join documentation, for IDs in y (docData) but not in x (authorData), an NA is returned. I went back to the CrosscheckData notebook to see how many IDs meeting the subset criteria were in docData but not authorData. There were exactly 47. Thus, it seems that the discrepancy is due to IDs that appeared in docData but not authorData. 

In [29]:
# Create vector of subset IDs in Y but not X.
missingIds  <- read.table('missingIds.txt')

In [30]:
missingIds  <- missingIds$x

In [31]:
# Confirm that these IDs are indeed in docData (y) but not in authorData (x)
summary(subset(docData, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "author_generation", "doctype")))
nrow(subset(authorData, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "author_generation")))

 docauthorid        author_generation    doctype         
 Length:48          Length:48          Length:48         
 Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character  

[1] 0

In [32]:
#There is one too many. Is this because a letter was written from outside North America?
subset(docData, grepl(paste(missingIds,collapse="|"), docauthorid), select = "wwritten")

wwritten                                                                                                                   
3382 Ebensburg, PA; Pennsylvania; United States; Mid-Atlantic States; Northeast States; East Coast States; North America        
3383 Albany, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America               
3386 United States; North America                                                                                               
3388 Racine, WI; Wisconsin; United States; East North Central States; Midwest States; Great Lakes States; North America         
3391 Cincinnati, OH; Ohio; United States; East North Central States; Midwest States; Mississippi Basin States; North America    
3393 North America                                                                                                              
3396 United States; North America                                                                                               
3398 Granville, OH; Ohio; United States; East North Central States; Midwest States; Mississippi Basin States; North America     
3400 New York, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America             
3402 Indiana; United States; East North Central States; Midwest States; Mississippi Basin States; North America                 
3405 Albany, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America               
3406 New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                           
3407 Utica, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                
3410 Utica, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                
3411 Utica, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                
3414 Utica, NY; New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                
3416 New York; United States; Mid-Atlantic States; Northeast States; East Coast States; North America                           
3418 Philadelphia, PA; Pennsylvania; United States; Mid-Atlantic States; Northeast States; East Coast States; North America     
3422 Cincinnati, OH; Ohio; United States; East North Central States; Midwest States; Mississippi Basin States; North America    
3440 Midwest States; United States; North America                                                                               
3441 Racine, WI; Wisconsin; United States; East North Central States; Midwest States; Great Lakes States; North America         
3442 Racine, WI; Wisconsin; United States; East North Central States; Midwest States; Great Lakes States; North America         
3448 Wales; United Kingdom; Western Europe; Europe                                                                              
3451 Tennessee; United States; Appalachian Mountain States; Southern States; Mississippi Basin States; North America            
3454 Missouri; United States; West North Central States; Midwest States; Mississippi Basin States; North America                
3456 Missouri; United States; West North Central States; Midwest States; Mississippi Basin States; North America                
3459 Emporia, KS; Kansas; United States; West North Central States; Midwest States; Great Plains States; North America          
3460 Emporia, KS; Kansas; United States; West North Central States; Midwest States; Great Plains States; North America          
3464 Kansas; United States; West North Central States; Midwest States; Great Plains States; North America                       
3469 Nebraska; United States; West North Central States; Midwest States; Great Plains States; North America                     
3477 O

Yes, one of these letters was written outside the United States. So we now have 47 letters associated with IDs in y but not x. Now, how do these appear after the merge (i.e., as NAs)? 

In [33]:
# This code does not work 
#df[df$docauthorid[missingIds],c("docauthorid", "author_generation.x", "author_generation.y")]

#But this code does seeem to indicate that the 47 letters were coded as NA for author_generation.x
subset(df, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "author_generation.x", "author_generation.y", "doctype"))
nrow(subset(df, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "author_generation.x", "author_generation.y", "doctype")))

docauthorid author_generation.x author_generation.y doctype
3382 per0031173  NA                  First               Letter 
3383 per0031175  NA                  First               Letter 
3386 per0041383  NA                  First               Letter 
3388 per0031180  NA                  First               Letter 
3391 per0004486  NA                  First               Letter 
3393 per0041388  NA                  First               Letter 
3396 per0041394  NA                  First               Letter 
3398 per0031224  NA                  First               Letter 
3400 per0031227  NA                  First               Letter 
3402 per0031228  NA                  First               Letter 
3405 per0031175  NA                  First               Letter 
3406 per0031251  NA                  First               Letter 
3407 per0031257  NA                  First               Letter 
3410 per0031263  NA                  First               Letter 
3411 per0032144  NA                  First               Letter 
3414 per0031268  NA                  First               Letter 
3416 per0031335  NA                  First               Letter 
3418 per0031337  NA                  First               Letter 
3422 per0031343  NA                  First               Letter 
3440 per0035349  NA                  First               Letter 
3441 per0031180  NA                  First               Letter 
3442 per0031180  NA                  First               Letter 
3448 per0004486  NA                  First               Letter 
3451 per0004486  NA                  First               Letter 
3454 per0041985  NA                  First               Letter 
3456 per0032161  NA                  First               Letter 
3459 per0033380  NA                  First               Letter 
3460 per0033381  NA                  First               Letter 
3464 per0004492  NA                  First               Letter 
3469 per0004495  NA                  First               Letter 
3477 per0031827  NA                  First               Letter 
3482 per0033407  NA                  First               Letter 
3496 per0035356  NA                  First               Letter 
3503 per0033412  NA                  First               Letter 
3507 per0042015  NA                  First               Letter 
3513 per0032241  NA                  First               Letter 
3514 per0032241  NA                  First               Letter 
3517 per0004521  NA                  First               Letter 
3518 per0004522  NA                  First               Letter 
3520 per0041996  NA                  First               Letter 
3521 per0032259  NA                  First               Letter 
3537 per0004526  NA                  First               Letter 
3547 per0004533  NA                  First               Letter 
3574 per0033256  NA                  First               Letter 
3577 per0033262  NA                  First               Letter 
3578 per0033267  NA                  First               Letter 
3579 per0033268  NA                  First               Letter 
3580 per0033271  NA                  First               Letter

[1] 48

In [34]:
filter(df, grepl(paste(missingIds,collapse="|"), docauthorid)) %>% select(ends_with("y"))

docauthorname.y                                   
1  Roberts, George, fl. 1850                         
2  Harry, David Shone, fl. 1817                      
3  Jones, Edward, fl. 1837                           
4  Cheshire, John, fl. 1847                          
5  Roberts, Samuel, fl. 1856-1870                    
6  Jenkins, William, fl. 1862                        
7  Morddal, fl. 1866                                 
8  Owen, John, fl. 1867                              
9  Lloyd, John, fl. 1868                             
10 James, Hiram                                      
11 Harry, David Shone, fl. 1817                      
12 Richards, John, fl. 1817                          
13 Thomas, William, fl. 1818                         
14 Davies, William, fl. 1794-1821                    
15 Lewis, John, fl. 1832                             
16 Evans, Hugh, fl. 1843                             
17 Howell, Lewis, Jr., fl. 1844                      
18 Jenkins, Samuel, 1789(?)-                         
19 James, John, fl. 1829                             
20 Pugh, Margaret                                    
21 Cheshire, John, fl. 1847                          
22 Cheshire, John, fl. 1847                          
23 Roberts, Samuel, fl. 1856-1870                    
24 Roberts, Samuel, fl. 1856-1870                    
25 James, William, fl. 1869                          
26 Evans, T. R., 1842(?)-                            
27 Anonymous Welsh Immigrant, D. W., fl. 1870        
28 Anonymous Welsh Farmer in Emporia, KS, fl. 1870   
29 Jones, Joseph, fl. 1872                           
30 Lewis, John Newell, fl. 1873                      
31 Thomas, David W., fl. 1884                        
32 Davies, David, fl. 1834                           
33 Phillips, Sarah, fl. 1869                         
34 Thomas, T., fl. 1873                              
35 Davies, David, fl. 1881                           
36 Williams, Robert, fl. 1832-1833                   
37 Williams, Robert, fl. 1832-1833                   
38 Morgan, Rees, fl. 1864                            
39 Davies, D. Tydvil, fl. 1868                       
40 Anonymous Welsh Tin-Plate Worker in Ohio, fl. 1873
41 Davies, David R., fl.1874                         
42 Griffiths, Thomas, fl. 1857                       
43 Joel, Lewis Francis, fl. 1873                     
44 Anonymous Welshman in Wisconsin, Evan, fl. 1865   
45 Jones, Dan, fl. 1849                              
46 Morgan, William, fl. 1849                         
47 Howell, Evan, fl. 1851                            
48 Evans, Samuel, fl. 1854                           
   briefname.y                              authrace.y nationality.y  
1  George Roberts                           White      American       
2  David Shone Harry                        White      NA             
3  Edward Jones                             White      NA             
4  John Cheshire                            White      American       
5  Samuel Roberts                           White      Welsh; European
6  William Jenkins                          White      NA             
7  Morddal                                  White      NA             
8  John Owen                                White      NA             
9  John Lloyd                               White      American       
10 Hiram James                              White      NA             
11 David Shone Harry                        White      NA             
12 John Richards                            White      NA             
13 William Thomas                           White      NA             
14 William Davies                           White      American       
15 John Lewis                               White      American       
16 Hugh Evans                               White      American       
17 Lewis Howell, Jr., fl. 1844              White      NA             
18 Samuel Jenkins                           White      American       
19 John 

These cases have missing data in authorData (x) but not entirely in docData (y).

## Recode the 48 cases above using y data

In [35]:
rows = which(grepl(paste(missingIds,collapse="|"), df$docauthorid))
length(rows)
print(rows)

[1] 48

 [1] 3382 3383 3386 3388 3391 3393 3396 3398 3400 3402 3405 3406 3407 3410 3411
[16] 3414 3416 3418 3422 3440 3441 3442 3448 3451 3454 3456 3459 3460 3464 3469
[31] 3477 3482 3496 3503 3507 3513 3514 3517 3518 3520 3521 3537 3547 3574 3577
[46] 3578 3579 3580


In [36]:
df$author_generation[rows] <- df$author_generation.y[rows]
df$author_generation[rows]

[1] First First First First First First First First First First First First
[13] First First First First First First First First First First First First
[25] First First First First First First First First First First First First
[37] First First First First First First First First First First First First
Levels: First Not applicable Other Second

In [37]:
subset(df, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "author_generation.x", "author_generation.y", "author_generation", "doctype"))

docauthorid author_generation.x author_generation.y author_generation
3382 per0031173  NA                  First               First            
3383 per0031175  NA                  First               First            
3386 per0041383  NA                  First               First            
3388 per0031180  NA                  First               First            
3391 per0004486  NA                  First               First            
3393 per0041388  NA                  First               First            
3396 per0041394  NA                  First               First            
3398 per0031224  NA                  First               First            
3400 per0031227  NA                  First               First            
3402 per0031228  NA                  First               First            
3405 per0031175  NA                  First               First            
3406 per0031251  NA                  First               First            
3407 per0031257  NA                  First               First            
3410 per0031263  NA                  First               First            
3411 per0032144  NA                  First               First            
3414 per0031268  NA                  First               First            
3416 per0031335  NA                  First               First            
3418 per0031337  NA                  First               First            
3422 per0031343  NA                  First               First            
3440 per0035349  NA                  First               First            
3441 per0031180  NA                  First               First            
3442 per0031180  NA                  First               First            
3448 per0004486  NA                  First               First            
3451 per0004486  NA                  First               First            
3454 per0041985  NA                  First               First            
3456 per0032161  NA                  First               First            
3459 per0033380  NA                  First               First            
3460 per0033381  NA                  First               First            
3464 per0004492  NA                  First               First            
3469 per0004495  NA                  First               First            
3477 per0031827  NA                  First               First            
3482 per0033407  NA                  First               First            
3496 per0035356  NA                  First               First            
3503 per0033412  NA                  First               First            
3507 per0042015  NA                  First               First            
3513 per0032241  NA                  First               First            
3514 per0032241  NA                  First               First            
3517 per0004521  NA                  First               First            
3518 per0004522  NA                  First               First            
3520 per0041996  NA                  First               First            
3521 per0032259  NA                  First               First            
3537 per0004526  NA                  First               First            
3547 per0004533  NA                  First               First            
3574 per0033256  NA                  First               First            
3577 per0033262  NA                  First               First            
3578 per0033267  NA                  First               First            
3579 per0033268  NA                  First               First            
3580 per0033271  NA                  First               First            
     doctype
3382 Letter 
3383 Letter 
3386 Letter 
3388 Letter 
3391 Letter 
3393 Letter 
3396 Letter 
3398 Letter 
3400 Letter 
3402 Letter 
3405 Letter 
3406 Letter 
3407 Letter 
3410 Letter 
3411 Letter 
3414 Letter 
3416 Letter 
3418 Letter 
3422 Letter 
3440 Letter 
3441 Letter 
3442 Letter 
3448 Letter 
3451 Letter 
3454 

In [38]:
summary(df$author_generation)
summary(df$author_generation.x)
summary(df$author_generation.y)

First Not applicable          Other         Second           NA's 
          6073            735             35            625           1281

First Not applicable          Other         Second           NA's 
          6025            735             35            625           1329

First Not applicable          Other         Second           NA's 
          6054            735             35            625           1300

In [39]:
#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, 
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, 
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation.x == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, 
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation.y == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 1272

[1] 1225

[1] 1272

To Do: Do the above for all key author level variables (i.e., gender, religion, cultural heritage, occupation, marriage status, parental status, stayed in North America). 

In [40]:
# Compare gender
summary(df$authorgender.x)
summary(df$authorgender.y)

F    M NA's 
3924 4632  193

F    M 
3942 4807

In [41]:
# Create new variable and populate with authorData (x)
df$authorgender <- df$authorgender.x
summary(df$authorgender)

F    M NA's 
3924 4632  193

In [42]:
# Check to make sure the rows with missing IDs are all NAs
subset(df, grepl(paste(missingIds,collapse="|"), docauthorid), select = c("docauthorid", "authorgender", "authorgender.x", "authorgender.y", "doctype"))

docauthorid authorgender authorgender.x authorgender.y doctype
3382 per0031173  NA           NA             M              Letter 
3383 per0031175  NA           NA             M              Letter 
3386 per0041383  NA           NA             M              Letter 
3388 per0031180  NA           NA             M              Letter 
3391 per0004486  NA           NA             M              Letter 
3393 per0041388  NA           NA             M              Letter 
3396 per0041394  NA           NA             M              Letter 
3398 per0031224  NA           NA             M              Letter 
3400 per0031227  NA           NA             M              Letter 
3402 per0031228  NA           NA             M              Letter 
3405 per0031175  NA           NA             M              Letter 
3406 per0031251  NA           NA             M              Letter 
3407 per0031257  NA           NA             M              Letter 
3410 per0031263  NA           NA             M              Letter 
3411 per0032144  NA           NA             M              Letter 
3414 per0031268  NA           NA             M              Letter 
3416 per0031335  NA           NA             M              Letter 
3418 per0031337  NA           NA             M              Letter 
3422 per0031343  NA           NA             M              Letter 
3440 per0035349  NA           NA             F              Letter 
3441 per0031180  NA           NA             M              Letter 
3442 per0031180  NA           NA             M              Letter 
3448 per0004486  NA           NA             M              Letter 
3451 per0004486  NA           NA             M              Letter 
3454 per0041985  NA           NA             M              Letter 
3456 per0032161  NA           NA             M              Letter 
3459 per0033380  NA           NA             M              Letter 
3460 per0033381  NA           NA             M              Letter 
3464 per0004492  NA           NA             M              Letter 
3469 per0004495  NA           NA             M              Letter 
3477 per0031827  NA           NA             M              Letter 
3482 per0033407  NA           NA             M              Letter 
3496 per0035356  NA           NA             F              Letter 
3503 per0033412  NA           NA             M              Letter 
3507 per0042015  NA           NA             M              Letter 
3513 per0032241  NA           NA             M              Letter 
3514 per0032241  NA           NA             M              Letter 
3517 per0004521  NA           NA             M              Letter 
3518 per0004522  NA           NA             M              Letter 
3520 per0041996  NA           NA             M              Letter 
3521 per0032259  NA           NA             M              Letter 
3537 per0004526  NA           NA             M              Letter 
3547 per0004533  NA           NA             M              Letter 
3574 per0033256  NA           NA             M              Letter 
3577 per0033262  NA           NA             M              Letter 
3578 per0033267  NA           NA             M              Letter 
3579 per0033268  NA           NA             M              Letter 
3580 per0033271  NA           NA             M              Letter

In [43]:
# Replace NA in missing data rows with value for docData (y)
df$authorgender[rows] <- df$authorgender.y[rows]
summary(df$authorgender)

F    M NA's 
3926 4678  145

In [44]:
summary(subset(df, is.na(authorgender), select = c("docauthorid", "authorgender", "authorgender.x", "authorgender.y", "doctype")))

     docauthorid  authorgender authorgender.x authorgender.y
 per0000339:  4   F   :  0     F   :  0       F: 15         
 per0004491:  3   M   :  0     M   :  0       M:130         
 per0004529:  3   NA's:145     NA's:145                     
 per0004552:  3                                             
 per0031346:  3                                             
 per0031824:  3                                             
 (Other)   :126                                             
             doctype   
 Letter          :143  
 Emigration guide:  2  
 Cartoon         :  0  
 Chapter         :  0  
 Diary           :  0  
 Editorial       :  0  
 (Other)         :  0  

This shows that there are still 145 NAs for authorgender variable. These may be cases that will not be included in the subset.

In [45]:
# Check number of cases for which gender is NOT NA and the other conditions are met ...
# ...to see if they match the case counts from the docauthorid / briefname review.
nrow(filter(df, !is.na(authorgender) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, !is.na(authorgender.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, !is.na(authorgender.y) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 1272

[1] 1225

[1] 1272

They do match. What this shows is that after replacing missing authorData values for gender with docData values, there are no NA cases that meet the conditions for migrant generation, language, format and place written. All appears to be good with this variable. Moving on.

In [46]:
# Compare cultural heritage
#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, is.na(cultural_heritage.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(cultural_heritage.y) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 48

[1] 1

In [47]:
# Create new variable and populate with authorData (x)
df$cultural_heritage <- df$cultural_heritage.x

# For IDs in docData (y) but not authorData (x), replace x value with y value.
df$cultural_heritage[rows] <- df$cultural_heritage.y[rows]

# Check amount of cultural heritage missing data in the eventual subset.

nrow(filter(df, is.na(cultural_heritage) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(cultural_heritage.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(cultural_heritage.y) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 1

[1] 48

[1] 1

In [48]:
glimpse(filter(df, is.na(cultural_heritage) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

Observations: 1
Variables: 50
$ docsequence                 <int> 19
$ docid                       <fct> S9873-D019
$ docyear                     <int> 1830
$ doctype                     <fct> Letter
$ allsubject                  <fct> Crops; Farms; Immigration and emigration;…
$ broadsubj                   <fct> Agriculture; Communities; Domestic life; …
$ personalevent               <fct> NA
$ wwritten                    <fct> Canada; North America
$ docauthorid                 <fct> per0038979
$ docauthorname.x             <fct> "Prongley, Esau, fl. 1830"
$ docauthorname.y             <fct> "Prongley, Esau, fl. 1830"
$ language                    <fct> English
$ editor                      <fct> NA
$ briefname.x                 <fct> Esau Prongley
$ briefname.y                 <fct> Esau Prongley
$ authrace.x                  <fct> White
$ authrace.y                  <fct> White
$ nationality.x               <fct> NA
$ nationality.y               <fct> NA
$ religion.x               

47 out of 48 cases of missing data for cultural heritage are resolved. Unable to confirm cultural heritage of remaining NA by looking at original text or searching online. Will leave as-is. This case will be omitted from some statistical procedures.

In [49]:
# Compare religion
#Subset data to isolate letters written by immigrants in North America
nrow(filter(df, is.na(religion.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(religion.y) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 405

[1] 398

In [50]:
# Create new variable and populate with authorData (x)
df$religion <- as.character(df$religion.x)
df$religion.y <- as.character(df$religion.y)

In [51]:
data.frame(df$religion[rows], df$religion.y[rows])

df.religion.rows. df.religion.y.rows.         
1  NA                NA                          
2  NA                NA                          
3  NA                NA                          
4  NA                NA                          
5  NA                Congregationalist; Christian
6  NA                NA                          
7  NA                NA                          
8  NA                NA                          
9  NA                NA                          
10 NA                NA                          
11 NA                NA                          
12 NA                NA                          
13 NA                NA                          
14 NA                Baptist; Christian          
15 NA                NA                          
16 NA                NA                          
17 NA                NA                          
18 NA                NA                          
19 NA                NA                          
20 NA                NA                          
21 NA                NA                          
22 NA                NA                          
23 NA                Congregationalist; Christian
24 NA                Congregationalist; Christian
25 NA                NA                          
26 NA                NA                          
27 NA                NA                          
28 NA                NA                          
29 NA                NA                          
30 NA                Christian                   
31 NA                NA                          
32 NA                NA                          
33 NA                NA                          
34 NA                NA                          
35 NA                NA                          
36 NA                NA                          
37 NA                NA                          
38 NA                NA                          
39 NA                NA                          
40 NA                NA                          
41 NA                NA                          
42 NA                NA                          
43 NA                NA                          
44 NA                NA                          
45 NA                Mormon; Christian           
46 NA                Mormon; Christian           
47 NA                NA                          
48 NA                Mormon; Christian

In [52]:
# For IDs in docData (y) but not authorData (x), replace x value with y value.
df$religion[rows] <- df$religion.y[rows]
data.frame(df$religion[rows], df$religion.y[rows])

df.religion.rows.            df.religion.y.rows.         
1  NA                           NA                          
2  NA                           NA                          
3  NA                           NA                          
4  NA                           NA                          
5  Congregationalist; Christian Congregationalist; Christian
6  NA                           NA                          
7  NA                           NA                          
8  NA                           NA                          
9  NA                           NA                          
10 NA                           NA                          
11 NA                           NA                          
12 NA                           NA                          
13 NA                           NA                          
14 Baptist; Christian           Baptist; Christian          
15 NA                           NA                          
16 NA                           NA                          
17 NA                           NA                          
18 NA                           NA                          
19 NA                           NA                          
20 NA                           NA                          
21 NA                           NA                          
22 NA                           NA                          
23 Congregationalist; Christian Congregationalist; Christian
24 Congregationalist; Christian Congregationalist; Christian
25 NA                           NA                          
26 NA                           NA                          
27 NA                           NA                          
28 NA                           NA                          
29 NA                           NA                          
30 Christian                    Christian                   
31 NA                           NA                          
32 NA                           NA                          
33 NA                           NA                          
34 NA                           NA                          
35 NA                           NA                          
36 NA                           NA                          
37 NA                           NA                          
38 NA                           NA                          
39 NA                           NA                          
40 NA                           NA                          
41 NA                           NA                          
42 NA                           NA                          
43 NA                           NA                          
44 NA                           NA                          
45 Mormon; Christian            Mormon; Christian           
46 Mormon; Christian            Mormon; Christian           
47 NA                           NA                          
48 Mormon; Christian            Mormon; Christian

In [53]:
df$religion  <- factor(df$religion)
glimpse(df$religion)

 Factor w/ 57 levels "Albanian Orthodox; Christian",..: 43 12 12 12 12 12 12 12 12 12 ...


In [54]:
# Check amount of cultural heritage missing data in the eventual subset.

nrow(filter(df, is.na(religion) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(religion.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

nrow(filter(df, is.na(religion.y) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten))) #Writing only from North America

[1] 398

[1] 405

[1] 398

In [55]:
# Get the docids for subset using religion.

docids.1 <- filter(df, is.na(religion) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten)) %>%
select(docid)
docids.1  <- docids.1$docid
length(docids.1)

[1] 398

In [56]:
# Get the docids for subset using religion.x

docids.2 <- filter(df, is.na(religion.x) &
       doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten)) %>%
select(docid)
docids.2  <- docids.2$docid
length(docids.2)

[1] 405

In [57]:
# Get IDs  in the larger set but not in the smaller set.
diffsReligion <- setdiff(docids.2, docids.1)
diffsReligion

[1] "S316-D012" "S316-D031" "S316-D072" "S316-D090" "S316-D198" "S316-D199"
[7] "S316-D201"

In [58]:
filter(df, is.na(religion.x) &
            grepl(paste(diffsReligion,collapse="|"), docid) &
            doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten)) %>% #Writing only from North America
summary()


  docsequence          docid      docyear                 doctype 
 Min.   : 12.0   S316-D012:1   Min.   :1821   Letter          :7  
 1st Qu.: 51.5   S316-D031:1   1st Qu.:1849   Cartoon         :0  
 Median : 90.0   S316-D072:1   Median :1854   Chapter         :0  
 Mean   :114.7   S316-D090:1   Mean   :1852   Diary           :0  
 3rd Qu.:198.5   S316-D198:1   3rd Qu.:1858   Editorial       :0  
 Max.   :201.0   S316-D199:1   Max.   :1873   Emigration guide:0  
                 (Other)  :1                  (Other)         :0  
                                                                                                                                                                                                                                                                                                                                                                                allsubject
 Canals; Churches; Economic conditions; Towns; Transportation; Religion; Economics; C

These documents are missing from the dataset. Ok to exclude them. To do: occupation, marriage status, parental status, stayed in North America.

In [59]:
letters  <- filter(df, 
            !grepl("S316", docid) & #Not the missing source
            doctype == "Letter" & #Only letters
            grepl("English", language) & #Originally in English or translated into English
            author_generation == "First" & #Only 1st generation migrants
            grepl("North America", wwritten)) %>% #Writing only from North America
    select(-c(author_generation.x,author_generation.y))
glimpse(letters)

Observations: 1,222
Variables: 49
$ docsequence                 <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 3…
$ docid                       <fct> S10003-D023, S10003-D024, S10003-D025, S1…
$ docyear                     <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838,…
$ doctype                     <fct> Letter, Letter, Letter, Letter, Letter, L…
$ allsubject                  <fct> "Childbirth; Church attendance; Cities; F…
$ broadsubj                   <fct> Health; Religion; Communities; Relationsh…
$ personalevent               <fct> NA, NA, NA, NA, NA, NA, NA, NA, Physical …
$ wwritten                    <fct> "Baltimore, MD; Maryland; United States; …
$ docauthorid                 <fct> per0022938, per0022938, per0022938, per00…
$ docauthorname.x             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ docauthorname.y             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ language                    <fct> English; German, English; German, English…
$ editor          

## Adjustments to the Subset

Recode values per CrosscheckData findings and decisions (search for term "Decision" in that script)

<b>Nationality</b>: The only correction needed is a chapter and therefore not required.

<b>North American Occupation</b>: Omit "'s wife" from the values.

In [61]:
# To what value does this apply
unique(letters$north_american_occupation.x[which(grepl("wife", letters$north_american_occupation.x))])

[1] "Homemaker; Physician's wife"         
[2] "Farmer's wife; Military wife; Writer"
[3] "Farmer's wife"                       
[4] "Farmer's wife; Servant"

In [62]:
letters$north_american_occupation  <-  str_remove_all(letters$north_american_occupation.x, "[:space:]wife")

In [63]:
letters$north_american_occupation  <-  str_remove_all(letters$north_american_occupation, "\'s")

In [64]:
letters$north_american_occupation  <-  str_remove_all(letters$north_american_occupation, "[:space:]personnel")

In [65]:
unique(letters$north_american_occupation.x)
unique(letters$north_american_occupation)

[1] "Homemaker; Physician's wife"                                       
 [2] "Nun; Social worker; Teacher"                                       
 [3] NA                                                                  
 [4] "Jeweler"                                                           
 [5] "Retail worker"                                                     
 [6] "Businessman; Retail worker"                                        
 [7] "Factory worker"                                                    
 [8] "Student"                                                           
 [9] "Farmer"                                                            
[10] "Servant"                                                           
[11] "Homemaker"                                                         
[12] "Tailor"                                                            
[13] "Artist"                                                            
[14] "Businessman"                                                       
[15] "Clergy"                                                            
[16] "Missionary; Retail worker"                                         
[17] "Seamstress"                                                        
[18] "Laborer"                                                           
[19] "Cook; Secretary; Teacher; Writer"                                  
[20] "Physician"                                                         
[21] "Farmer's wife; Military wife; Writer"                              
[22] "Missionary"                                                        
[23] "Farmer's wife"                                                     
[24] "Architect; Artist; Businessman; Clergy; Educator"                  
[25] "Architect; Artist; Businessman; Clergy"                            
[26] "Architect; Artist; Businessman; Clergy; Educator; Farmer"          
[27] "Architect; Artist; Businessman; Educator; Farmer"                  
[28] "Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant"
[29] "Farmer's wife; Servant"                                            
[30] "Banker; Businessman; Engineer; Miner; Politician"                  
[31] "Government appointee; Politician; Religious leader"                
[32] "Farmer; Tradesman"                                                 
[33] "Miner; Rancher; Laborer"                                           
[34] "Clergy; Teacher"                                                   
[35] "Clergy; Educator; Writer"                                          
[36] "Tradesman"                                                         
[37] "Factory worker; Military personnel"                                
[38] "Farmer; Miner"                                                     
[39] "Teacher"                                                           
[40] "Surveyor"                                                          
[41] "Manufacturer"                                                      
[42] "Clergy; Missionary"                                                
[43] "Royal governor"                                                    
[44] "Laborer; Tradesman"                                                
[45] "Writer"                                                            
[46] "Miner"                                                             
[47] "Writer; Laborer"                                                   
[48] "Secretary; Accountant"                                             
[49] "Editor"                                                            
[50] "Cook; Housekeeper"                                                 
[51] "Nurse"                                                             
[52] "Religious worker"                                                  
[53] "Miner; Explorer"                                                   
[54] "Editor; Publisher; Writer"                                         
[55] 

[1] "Homemaker; Physician"                                              
 [2] "Nun; Social worker; Teacher"                                       
 [3] NA                                                                  
 [4] "Jeweler"                                                           
 [5] "Retail worker"                                                     
 [6] "Businessman; Retail worker"                                        
 [7] "Factory worker"                                                    
 [8] "Student"                                                           
 [9] "Farmer"                                                            
[10] "Servant"                                                           
[11] "Homemaker"                                                         
[12] "Tailor"                                                            
[13] "Artist"                                                            
[14] "Businessman"                                                       
[15] "Clergy"                                                            
[16] "Missionary; Retail worker"                                         
[17] "Seamstress"                                                        
[18] "Laborer"                                                           
[19] "Cook; Secretary; Teacher; Writer"                                  
[20] "Physician"                                                         
[21] "Farmer; Military; Writer"                                          
[22] "Missionary"                                                        
[23] "Architect; Artist; Businessman; Clergy; Educator"                  
[24] "Architect; Artist; Businessman; Clergy"                            
[25] "Architect; Artist; Businessman; Clergy; Educator; Farmer"          
[26] "Architect; Artist; Businessman; Educator; Farmer"                  
[27] "Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant"
[28] "Farmer; Servant"                                                   
[29] "Banker; Businessman; Engineer; Miner; Politician"                  
[30] "Government appointee; Politician; Religious leader"                
[31] "Farmer; Tradesman"                                                 
[32] "Miner; Rancher; Laborer"                                           
[33] "Clergy; Teacher"                                                   
[34] "Clergy; Educator; Writer"                                          
[35] "Tradesman"                                                         
[36] "Factory worker; Military"                                          
[37] "Farmer; Miner"                                                     
[38] "Teacher"                                                           
[39] "Surveyor"                                                          
[40] "Manufacturer"                                                      
[41] "Clergy; Missionary"                                                
[42] "Royal governor"                                                    
[43] "Laborer; Tradesman"                                                
[44] "Writer"                                                            
[45] "Miner"                                                             
[46] "Writer; Laborer"                                                   
[47] "Secretary; Accountant"                                             
[48] "Editor"                                                            
[49] "Cook; Housekeeper"                                                 
[50] "Nurse"                                                             
[51] "Religious worker"                                                  
[52] "Miner; Explorer"                                                   
[53] "Editor; Publisher; Writer"                                         
[54] "Merchant"                                                          
[55] 

In [67]:
# Convert strings back to factor
letters <- factorize(letters)
glimpse(letters)

Observations: 1,222
Variables: 50
$ docsequence                 <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 3…
$ docid                       <fct> S10003-D023, S10003-D024, S10003-D025, S1…
$ docyear                     <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838,…
$ doctype                     <fct> Letter, Letter, Letter, Letter, Letter, L…
$ allsubject                  <fct> "Childbirth; Church attendance; Cities; F…
$ broadsubj                   <fct> Health; Religion; Communities; Relationsh…
$ personalevent               <fct> NA, NA, NA, NA, NA, NA, NA, NA, Physical …
$ wwritten                    <fct> "Baltimore, MD; Maryland; United States; …
$ docauthorid                 <fct> per0022938, per0022938, per0022938, per00…
$ docauthorname.x             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ docauthorname.y             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ language                    <fct> English; German, English; German, English…
$ editor          

In [68]:
summary(letters)

  docsequence             docid         docyear       doctype    
 Min.   :  2.00   S10003-D023:   1   Min.   :1804   Letter:1222  
 1st Qu.: 25.00   S10003-D024:   1   1st Qu.:1858                
 Median : 54.00   S10003-D025:   1   Median :1868                
 Mean   : 72.47   S10003-D026:   1   Mean   :1881                
 3rd Qu.:110.00   S10003-D027:   1   3rd Qu.:1909                
 Max.   :239.00   S10003-D028:   1   Max.   :1979                
                  (Other)    :1216   NA's   :21                  
                                                         allsubject  
 Correspondence; Intellectual life                            :  10  
 Internment camps; Politics                                   :   8  
 Winter; Environment                                          :   6  
 Correspondence; Internment camps; Intellectual life; Politics:   5  
 Photographs; Intellectual life                               :   5  
 Business; Sons; Economics; Relationships           

In [69]:
write.csv(letters, "20201005_AM_FinalMeta.csv")

## Adjustments to the Subset

In [5]:
letters  <- read.csv("20201005_AM_FinalMeta.csv")
glimpse(letters)

Observations: 1,222
Variables: 51
$ X                           <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13…
$ docsequence                 <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 3…
$ docid                       <fct> S10003-D023, S10003-D024, S10003-D025, S1…
$ docyear                     <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838,…
$ doctype                     <fct> Letter, Letter, Letter, Letter, Letter, L…
$ allsubject                  <fct> "Childbirth; Church attendance; Cities; F…
$ broadsubj                   <fct> Health; Religion; Communities; Relationsh…
$ personalevent               <fct> NA, NA, NA, NA, NA, NA, NA, NA, Physical …
$ wwritten                    <fct> "Baltimore, MD; Maryland; United States; …
$ docauthorid                 <fct> per0022938, per0022938, per0022938, per00…
$ docauthorname.x             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ docauthorname.y             <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette,…
$ language        

In [6]:
letters[letters$docid == "S1019-D002", ]

X   docsequence docid      docyear doctype
137 137 2           S1019-D002 1872    Letter 
    allsubject                                                                               
137 Clergy; Missionaries; Students; Travel; Religion; Education; Entertainment and recreation
    broadsubj                                         personalevent
137 Religion; Education; Entertainment and recreation NA           
    wwritten                                                                                               
137 Ohio; United States; East North Central States; Midwest States; Mississippi Basin States; North America
    docauthorid ⋯ stayed_north_america.x stayed_north_america.y agewriting
137 per0001043  ⋯ Stayed                 Yes                    22        
    marriagestatus maternalstatus author_generation authorgender
137 Single         Childless      First             F           
    cultural_heritage religion            north_american_occupation  
137 Italian; European Catholic; Christian Nun; Social worker; Teacher

In [7]:
letters[letters$docauthorid == "per0001043", ]

X   docsequence docid      docyear doctype
137 137  2          S1019-D002 1872    Letter 
138 138  4          S1019-D004 1872    Letter 
139 139  5          S1019-D005 1872    Letter 
140 140  6          S1019-D006 1872    Letter 
141 141  7          S1019-D007 1873    Letter 
142 142  8          S1019-D008 1873    Letter 
143 143  9          S1019-D009 1873    Letter 
144 144 10          S1019-D010 1874    Letter 
145 145 11          S1019-D011 1874    Letter 
146 146 12          S1019-D012 1876    Letter 
147 147 13          S1019-D013 1876    Letter 
148 148 14          S1019-D014 1876    Letter 
149 149 15          S1019-D015 1877    Letter 
150 150 16          S1019-D016 1878    Letter 
151 151 17          S1019-D017 1878    Letter 
152 152 18          S1019-D018 1878    Letter 
153 153 19          S1019-D019 1879    Letter 
154 154 20          S1019-D020 1879    Letter 
155 155 21          S1019-D021 1879    Letter 
156 156 22          S1019-D022 1880    Letter 
157 157 23          S1019-D023 1880    Letter 
158 158 24          S1019-D024 1880    Letter 
159 159 25          S1019-D025 1880    Letter 
160 160 26          S1019-D026 1880    Letter 
161 161 27          S1019-D027 1880    Letter 
162 162 28          S1019-D028 1880    Letter 
163 163 29          S1019-D029 1881    Letter 
164 164 30          S1019-D030 1881    Letter 
165 165 31          S1019-D031 1881    Letter 
166 166 32          S1019-D032 1881    Letter 
167 167 33          S1019-D033 1881    Letter 
168 168 34          S1019-D034 1882    Letter 
169 169 35          S1019-D035 1882    Letter 
170 170 36          S1019-D036 1882    Letter 
171 171 37          S1019-D037 1882    Letter 
172 172 38          S1019-D038 1883    Letter 
173 173 39          S1019-D039 1883    Letter 
174 174 40          S1019-D040 1883    Letter 
175 175 41          S1019-D041 1884    Letter 
176 176 42          S1019-D042 1884    Letter 
177 177 43          S1019-D043 1885    Letter 
178 178 44          S1019-D044 1884    Letter 
179 179 45          S1019-D045 1884    Letter 
180 180 46          S1019-D046 1885    Letter 
181 181 47          S1019-D047 1885    Letter 
182 182 48          S1019-D048 1885    Letter 
183 183 49          S1019-D049 1886    Letter 
184 184 50          S1019-D050 1886    Letter 
185 185 51          S1019-D051 1886    Letter 
186 186 52          S1019-D052 1886    Letter 
187 187 53          S1019-D053 1886    Letter 
188 188 54          S1019-D054 1887    Letter 
189 189 55          S1019-D055 1889    Letter 
190 190 56          S1019-D056 1889    Letter 
191 191 57          S1019-D057 1891    Letter 
192 192 58          S1019-D058 1892    Letter 
    allsubject                                                                                                                                                                                                                              
137 Clergy; Missionaries; Students; Travel; Religion; Education; Entertainment and recreation                                                                                                                                               
138 Missionaries; Nuns; Railroad trips; Religious festivals; Travel incidents; Travelers; Religion; Entertainment and recreation                                                                                                            
139 Houses; Missionaries; Schools; Stagecoach travel; Travelers; Domestic life; Religion; Education; Transportation; Entertainment and recreation                                                                                           
140 Churches; Clubs; Discipline; Educational philosophies; Missionaries; Schools; Students; Religion; Entertainment and recreation; Education                                                                                               
141 Accidents; Missionaries; Murder; Religious beliefs; Students; Travelers; Transportation; Religion; Law; Education; Entertainment and rec

This record was throwing errors in getDataNAIL because it had been overwritten by another file. This has been corrected.